### PR TITLE
Optimize DataSetWithoutTimeGenerator with primitive long heap/set

### PR DESCRIPTION
**Pull request by hongzhi-gao**
## Summary

Optimize the multi-way merge hot path in `DataSetWithoutTimeGenerator` by replacing `PriorityQueue<Long>` / `HashSet<Long>` with trimmed, fastutil 8.5.8–derived primitive containers, avoiding autoboxing of timestamps and the related CPU/GC cost.

CPU profiles showed the hotspot on `HashSet.contains` → `HashMap.getNode` via `timeHeapPut`, which is hit frequently in workloads such as Pipe extraction.

img_v3_0214g_8b329c57-96ab-4599-9fd7-6e86c133b31g

## Changes

- Add `LongHeapPriorityQueue` and `LongOpenHashSet` (forked from fastutil, APIs trimmed to what TsFile needs; key methods annotated with upstream origins)
- Switch `DataSetWithoutTimeGenerator` to these containers, with capacity pre-sized by series count
- Update root `NOTICE` for the fastutil-derived code and copyrights
- Add regression tests:
  - `DataSetWithoutTimeGeneratorTest`: sparse multi-way merge and cross-batch continuation
  - `LongOpenHashSetAndHeapPriorityQueueTest`: heap/set correctness (including edge cases), parity vs boxed JDK collections, combined dedup-merge pattern, and a micro-benchmark

## Performance results

Micro-benchmark of the same heap+set dedup-merge pattern used by `DataSetWithoutTimeGenerator` (**50 series × 100,000 rows**), comparing JDK `PriorityQueue<Long>+HashSet<Long>` with the primitive implementation:

| Implementation | Time | Speedup |
|----------------|------|---------|
| boxed (`PriorityQueue` + `HashSet`) | ~40–55 ms | 1.0x |
| primitive (`LongHeapPriorityQueue` + `LongOpenHashSet`) | ~9 ms | **~4.5x–6.0x** |

Notes: micro-benchmarks vary with machine load; local runs typically land around **~2x–6x**. Absolute gain depends on series count and total points. Correctness tests vs boxed behavior all passed.

## Test plan

- [x] `mvn -pl tsfile test -Dtest=LongOpenHashSetAndHeapPriorityQueueTest,DataSetWithoutTimeGeneratorTest`
- [ ] Review fastutil fork scope and `LICENSE` compliance
- [ ] (Optional) Re-profile IoTDB Pipe extraction and confirm the `HashSet.contains` hotspot shrinks

### CODE REVIEW

File: java/tsfile/src/main/java/org/apache/tsfile/read/query/dataset/DataSetWithoutTimeGenerator.java [hunk 1]
```diff
@@ -141,12 +141,12 @@ public RowRecord nextWithoutConstraint() throws IOException {
   private void timeHeapPut(long time) {
     if (!timeSet.contains(time)) {
       timeSet.add(time);
-      timeHeap.add(time);
+      timeHeap.enqueue(time);
     }
```
[the remark this answers is not in the document]
Comment (hongzhi-gao) [reply java/tsfile/src/main/java/org/apache/tsfile/read/query/dataset/DataSetWithoutTimeGenerator.java:144]: Agreed, switched to if (timeSet.add(time)) to avoid the double probe.

File: java/tsfile/src/test/java/org/apache/tsfile/utils/LongOpenHashSetAndHeapPriorityQueueTest.java [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+   */
+  @Test
+  public void testPrimitiveStructuresPerformance() {
+    final int series = 50;
+    final int rows = 100_000;
+    final long baseTime = 1_700_000_000_000L;
+
+    runBoxed(series, 10_000, baseTime);
+    runPrimitive(series, 10_000, baseTime);
+
```
[the remark this answers is not in the document]
Comment (hongzhi-gao) [reply java/tsfile/src/test/java/org/apache/tsfile/utils/LongOpenHashSetAndHeapPriorityQueueTest.java:275]: Agreed, gated with Assume.assumeTrue behind -Dtsfile.runPerformanceTests=true, same as other TsFile perf tests.

File: NOTICE [hunk 3]
```diff
@@ -20,3 +20,10 @@ Copyright 2001-2019 The Apache Software Foundation
 
 This product includes software developed at
 The Apache Software Foundation (http://www.apache.org/).
+
+============================================================================
+
+This product includes code derived from fastutil (http://fastutil.di.unimi.it/):
+  Copyright (C) 2002-2022 Sebastiano Vigna
+  Copyright (C) 2003-2022 Paolo Boldi and Sebastiano Vigna
+  Licensed under the Apache License, Version 2.0
```
Comment (JackieTien97) [inline NOTICE]: no need to add this

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

File: NOTICE (hunk 3 above)
Comment (hongzhi-gao) [reply NOTICE]: Removed

**Review by ColinLeeo — APPROVED** [review]
LGTM
